src/theory/arith/arith_proof_utilities.cpp

```diff
@@ -44,6 +45,71 @@ std::vector<Node> getMacroSumUbCoeff(const std::vector<Pf>& pfs,
   return ret;
 }
 
+Node expandMacroSumUb(const std::vector<Node>& children,
+                      const std::vector<Node>& args,
+                      CDProof* cdp)
+{
+  if (TraceIsOn("macro::arith"))
+  {
+    Trace("macro::arith") << "Expand MACRO_ARITH_SCALE_SUM_UB" << std::endl;
+    for (const auto& child : children)
+    {
+      Trace("macro::arith") << "  child: " << child << std::endl;
+    }
+    Trace("macro::arith") << "   args: " << args << std::endl;
+  }
+  Assert(args.size() == children.size());
+  NodeManager* nm = NodeManager::currentNM();
+  ProofStepBuffer steps{cdp->getManager()->getChecker()};
+
+  // Scale all children, accumulating
+  std::vector<Node> scaledRels;
+  Node one = nm->mkConstInt(Rational(1));
+  for (size_t i = 0; i < children.size(); ++i)
+  {
+    TNode child = children[i];
+    TNode scalar = args[i];
+    /*
+    if (scalar==one)
+    {
+      // if scaled by one, just take original
+      scaledRels.push_back(child);
+      continue;
+    }
+    */
```

Comment:
Why is this disabled?

Reply:
I'll revert, its an optimization I'm working on in a dev branch.